redis/cache/redispipebp/monitored.go

```diff
@@ -54,6 +58,26 @@ func extractCommand(cmd string) string {
 	return cmd
 }
 
+func readGaugeValue(g prometheus.Gauge) float64 {
+	var m *io_prometheus_client.Metric
+	err := g.Write(m)
+	if err != nil {
+		// How should I handle failure here?
```

Comment:
I believe we would need to have a metric-related logger that is configurable. Is this the first time we are handling Prometheus related write errors? :thinking:

Reply:
Any pointers to how I would do that ?

Reply:
Good question. For example in the `filewatcher` package we pass in a `log.Wrapper`, see [here](https://github.com/reddit/baseplate.go/blob/master/filewatcher/filewatcher.go#L103)